src/bin/pg_autoctl/loop.c

```diff
@@ -216,6 +217,19 @@ keeper_service_run(Keeper *keeper, pid_t *start_pid)
 			}
 		}
 
+
+		if (!reportPgIsRunning && config->exit_keeper_if_postgres_not_running)
```

Comment:
In a non-intuitive way I think I would prefer this test to be split in two as in the following example. It allows to have comments at the right places, make it obvious that we do something specific when `config->exit_keeper_if_postgres_not_running` is in use, and allows us to also add things to that case pretty easily, in the future, independently of `reportPgIsRunning`.

``` C
		if (config->exit_keeper_if_postgres_not_running)
		{
			/*
			 * 
			 */
			if (!reportPgIsRunning)
			{
				...
			}
		}
```